README.md

```diff
@@ -1,98 +1,422 @@
+Macro Polo is designed for unit testing template macros. Macro Polo 
+is meant to make it easier to express tests of the resulting HTML of 
+individual template macros (units) within a specific context or with
+specific inputs. Unit testing macros tests them individually in 
+isolation. 
 
-## Getting help
+If you're looking at a template and want to test JavaScript-related 
+behavior that occurs within that template's resulting HTML, you want 
+to investigate functional testing tools and frameworks.
 
-Instruct users how to get help with this software; this might include links to an issue tracker, wiki, mailing list, etc.
+## Using Macro Polo
 
-**Example**
+### Quickstart
 
-If you have questions, concerns, bug reports, etc, please file an issue in this repository's Issue Tracker.
+Create a Python file with the rest of your test suite, such as
+`test_templates.py`. This file will need to define a 
+[base test case](#creating-a-base-testcase-class),
+load template tests from JSON, and (optionally) use Python's
+`unittest.main()` to [run the tests](#running-tests):
 
-## Getting involved
+```python
+from macropolo import MacroTestCase, Jinja2Environment
+from macropolo import JSONTestCaseLoader
 
-This section should detail why people should get involved and describe key areas you are
-currently focusing on; e.g., trying to get feedback on features, fixing certain bugs, building
-important pieces, etc.
+class MyBaseTestCase(Jinja2Environment, MacroTestCase):
+    """
+    A MacroTestCase subclass for my Jinja2 Templates.
+    """
 
-General instructions on _how_ to contribute should be stated with a link to [CONTRIBUTING](CONTRIBUTING.md).
+    def search_root(self):
+        """
+        Return the root of the search path for templates.
+        """
+        # If the tests live under 'site_root/tests'...
+        root_dir = os.path.abspath(os.path.join(os.path.dirname( __file__ ),
+                                                os.pardir))
+        return root_dir
 
+    def search_exceptions(self):
+        """
+        Return a list of a subdirectory names that should not be searched
+        for templates.
+        """
+        return ['tests',]
+
+# Create MyTestCase subclasses for all JSON tests and add them to the
+# module's global context.
+tests_path = os.path.abspath(os.path.join(os.path.dirname( __file__ ), 'template_tests'))
+JSONTestCaseLoader(tests_path, MyBaseTestCase, globals())
+
+# Run the tests if we're executed
+if __name__ == '__main__':
+    unittest.main()
+```
+
+Then create your [JSON test specifications](#defining-tests-in-json) in
+the `template_tests` subdirectory of `tests`.
+
+### Creating a Base TestCase Class
+
+Test Case classes should inherit from a
+[template environment mixin](#template-environment-mixins), the
+[`MacroTestCase`](#macrotestcase) class, and should provide the 
+following methods:
+
+##### `search_root()`
+
+Return the root of the search path for templates.
+
+##### `search_exceptions()`
```

Comment:
Ah, yes. And it looks like that's the case for the other methods documented further down. Perils of changing your mind about heading levels after most of the content is written :)